include/ps/internal/message.h

```diff
@@ -90,6 +107,8 @@ struct Node {
   std::string hostname;
   /** \brief the port this node is binding */
   int port;
+  /** \brief whether this node is created by failover */
+  std::atomic<bool> is_recovery;
```

Comment:
what is_recovery should be atomic? if we don't do parallel read and write, then it's not necessary.